### PR TITLE
STR 3717 add snark acct inbox range rpc

### PR DESCRIPTION
**Pull request by krsnapaudel**
## Description

Adds `strata_getSnarkAcctInboxMsgRange(account_id, start, end)` to `OLClientRpc`.
The range is half-open `[start, end)`. The RPC returns
`Vec<RpcIndexedEntry<RpcMessageEntry>>`, pairing each Snark account inbox message
with its absolute inbox index.

The implementation reuses the existing
`OLRpcProvider::get_account_inbox_messages` storage/MMR path.

Changes:
- `crates/ol/rpc/types/src/account_summary.rs`: add generic `RpcIndexedEntry<T>` (`Serialize` / `Deserialize` / `JsonSchema`).
- `crates/ol/rpc/types/src/lib.rs`: re-export `RpcIndexedEntry`.
- `crates/ol/rpc/api/src/lib.rs`: add `getSnarkAcctInboxMsgRange` to `OLClientRpc` with `[start, end)` semantics.
- `bin/strata/src/rpc/node.rs`: validate `start <= end`, call the provider, and enumerate results from `start`. Includes a debug_assert! against storage range regressions and delegates unknown-account behavior to the existing storage/MMR lookup.
- `bin/strata/src/rpc/node_tests.rs`: add tests for valid range, empty range, reversed range, and provider error mapping.

This PR was created with help from Codex and Claude Code.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactor
- [ ] New or updated tests
- [ ] Dependency Update

## Notes to Reviewers

Is this PR addressing any specification, design doc or external reference document?

- [ ]  Yes
- [x]  No

If yes, please add relevant links:

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added (where necessary) tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.
- [x] I have [disclosed my use of AI](https://github.com/alpenlabs/alpen/blob/main/CONTRIBUTING.md#ai-assistance-notice) in the body of this PR.

## Related Issues

[STR-3717](https://alpenlabs.atlassian.net/browse/STR-3717)

[STR-3717]: https://alpenlabs.atlassian.net/browse/STR-3717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: crates/ol/rpc/types/src/account_summary.rs [hunk 1]
```diff
@@ -177,6 +177,33 @@ impl From<UpdateInputData> for RpcUpdateInputData {
     }
 }
 
+/// RPC serializable value with its absolute index.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "jsonschema", derive(schemars::JsonSchema))]
+pub struct RpcIndexedEntry<T> {
```
Comment (delbonis) [inline crates/ol/rpc/types/src/account_summary.rs:183]: I thought this existed already in some generic RPC API types module?

**Review by delbonis — APPROVED** [review]
Pretty minor comment, approving otherwise.

File: crates/ol/rpc/types/src/account_summary.rs (hunk 1 above)
Comment (krsnapaudel) [reply crates/ol/rpc/types/src/account_summary.rs:183]: @delbonis we have this: `strata_primitives::indexed::Indexed<T, Idx>`. No rpc type.

**Review by purusang — APPROVED** [review]

File: bin/strata/src/rpc/node_tests.rs [hunk 2]
```diff
@@ -528,6 +528,22 @@ fn rpc_messages_to_entries(messages: &[RpcMessageEntry]) -> Vec<MessageEntry> {
...
+    messages: &[RpcIndexedEntry<RpcMessageEntry>],
+) -> Vec<(u64, MessageEntry)> {
+    messages
+        .iter()
+        .map(|msg| {
+            let entry = msg
+                .value()
+                .clone()
+                .try_into()
+                .expect("message payload bytes must fit within SSZ max length");
```
Comment (storopoli) [inline bin/strata/src/rpc/node_tests.rs:541]: @delbonis would this be refactored after the `SszDelegate` thing is merged?

**Review by storopoli — APPROVED** [review]
ACK 2563d1ac0bc81b2b35d3c9370cd3dee0eb41659c

File: bin/strata/src/rpc/node_tests.rs (hunk 2 above)
Comment (storopoli) [reply bin/strata/src/rpc/node_tests.rs:541]: Solving this to put in the queue.
